src/js/components/conversation.jsx

```diff
@@ -54,34 +66,146 @@ export class ConversationComponent extends Component {
 
     };
 
+    onScroll = () => {
+        const {dispatch, shouldScrollToBottom} = this.props;
+
+        // If top of Conversation component is reached, we need to fetch older messages
+        const node = findDOMNode(this);
+        if (node.scrollTop === 0) {
+            this.fetchHistory();
+        } else if (shouldScrollToBottom) {
+            // Once we've started scrolling, we don't want the default behavior to force the scroll to the bottom afterwards
+            dispatch(setShouldScrollToBottom(false));
```

Comment:
`onScroll` get debounced and fires every 200ms on non-mobile devices

I had to do it this way because the mobile scroll was behaving really weirdly.